src/machi_cr_client.erl

```diff
@@ -935,3 +923,22 @@ timeout(infinity) ->
     timeout(15*60*1000);                        % close enough to infinity
 timeout(Timeout0) ->
     {Timeout0, Timeout0 + 30*1000}.
+
+trim_both_side([], _Offset, _Size) -> [];
+trim_both_side([{F, Offset0, Chunk, _Csum}|L], Offset, Size)
+  when Offset0 < Offset ->
+    TrashLen = 8 * (Offset - Offset0),
+    <<_:TrashLen/binary, NewChunk/binary>> = Chunk,
+    NewH = {F, Offset, NewChunk, <<>>},
+    trim_both_side([NewH|L], Offset, Size);
+trim_both_side(Chunks, Offset, Size) ->
+    %% TODO: optimize
+    [{F, Offset1, Chunk1, _Csum1}|L] = lists:reverse(Chunks),
+    Size1 = iolist_size(Chunk1),
+    if Offset + Size < Offset1 + Size1 ->
```

Comment:
I would prefer clearer variable names like `Position` and `Length` instead of `Offset1` and `Size1` for this computation. I think that will make things easier to understand for people that come after us. It might be worthwhile to comment the code to make its purpose clear.